srecord/input/file/aomf.cc

```diff
@@ -115,9 +115,9 @@ o96name(int x) -> const char *
 
 srecord::input_file_aomf::~input_file_aomf()
 {
-    if (current_buffer) {
```

Comment:
A holdover from earlier times when this wasn't guaranteed and habbits from embedded systems to never trust a null ptr.

I'm ok with including this check (and changing to suit)

Reply:
The nullptr checks were only removed where `delete` is called. That's safe.